pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,313 @@
+	Bytesoutcounter                  int           `json:"bytesOutCounter"`
+	Msgoutcounter                    int           `json:"msgOutCounter"`
+	Msgrateredeliver                 float64       `json:"msgRateRedeliver"`
+	Chuckedmessagerate               int           `json:"chuckedMessageRate"`
+	Msgbacklog                       int           `json:"msgBacklog"`
+	Msgbacklognodelayed              int           `json:"msgBacklogNoDelayed"`
+	Blockedsubscriptiononunackedmsgs bool          `json:"blockedSubscriptionOnUnackedMsgs"`
+	Msgdelayed                       int           `json:"msgDelayed"`
+	Unackedmessages                  int           `json:"unackedMessages"`
+	Type                             string        `json:"type"`
+	Msgrateexpired                   float64       `json:"msgRateExpired"`
+	Lastexpiretimestamp              int           `json:"lastExpireTimestamp"`
+	Lastconsumedflowtimestamp        int64         `json:"lastConsumedFlowTimestamp"`
+	Lastconsumedtimestamp            int           `json:"lastConsumedTimestamp"`
+	Lastackedtimestamp               int           `json:"lastAckedTimestamp"`
+	Consumers                        []interface{} `json:"consumers"`
+	Isdurable                        bool          `json:"isDurable"`
+	Isreplicated                     bool          `json:"isReplicated"`
+	Consumersaftermarkdeleteposition struct {
+	} `json:"consumersAfterMarkDeletePosition"`
+}
+
+type pulsarStats struct {
+	Msgratein         float64                       `json:"msgRateIn"`
+	Msgthroughputin   float64                       `json:"msgThroughputIn"`
+	Msgrateout        float64                       `json:"msgRateOut"`
+	Msgthroughputout  float64                       `json:"msgThroughputOut"`
+	Bytesincounter    int                           `json:"bytesInCounter"`
+	Msgincounter      int                           `json:"msgInCounter"`
+	Bytesoutcounter   int                           `json:"bytesOutCounter"`
+	Msgoutcounter     int                           `json:"msgOutCounter"`
+	Averagemsgsize    float64                       `json:"averageMsgSize"`
+	Msgchunkpublished bool                          `json:"msgChunkPublished"`
+	Storagesize       int                           `json:"storageSize"`
+	Backlogsize       int                           `json:"backlogSize"`
+	Publishers        []interface{}                 `json:"publishers"`
+	Subscriptions     map[string]pulsarSubscription `json:"subscriptions"`
+	Replication       struct {
+	} `json:"replication"`
+	Deduplicationstatus string `json:"deduplicationStatus"`
+}
+
+// NewPulsarScaler creates a new PulsarScaler
+func NewPulsarScaler(config *ScalerConfig) (Scaler, error) {
+	pulsarMetadata, err := parsePulsarMetadata(config)
+	if err != nil {
+		pulsarLog.Error(err, "error parsing pulsar metadata")
+		return nil, fmt.Errorf("error parsing pulsar metadata: %s", err)
+	}
+
+	client := &http.Client{}
```

Comment:
Please, use [the internal function to generate the httpClient](https://github.com/kedacore/keda/blob/main/pkg/util/http.go#L35), we use it to unify all the configurations. You can find an example [here](https://github.com/kedacore/keda/blob/main/pkg/scalers/rabbitmq_scaler.go#L116)